src/services/users.service.js

```diff
@@ -0,0 +1,46 @@
+let users = [];
+let userIdCounter = 1;
+
+const resetUsers = () => {
+  users = [];
+};
+
+const getAllUsers = () => {
+  return users;
+};
+
+const getUserById = (id) => {
+  return users.find((user) => user.id === Number(id)) || null;
+};
+
+const createUser = (name) => {
+  const user = {
+    name,
+    id: userIdCounter++,
+  };
+
+  users.push(user);
+
+  return user;
+};
+
+const deleteUser = (id) => {
+  users = users.filter((user) => user.id !== Number(id));
+};
+
+const updateUser = ({ id, name }) => {
+  const user = getUserById(id);
+
+  Object.assign(user, { name });
```

Comment:
The `updateUser` function currently only updates the `name` of a user. If other fields are intended to be updated, they should be included in the `Object.assign` call.